src/System.Memory/tests/ReadOnlySpan/StartsWith.StringComparison.cs

```diff
@@ -251,89 +251,6 @@ public static void StartsWithNoMatchNonOrdinal_StringComparison()
-        [InlineData("", "", StringComparison.Ordinal, true)]
-        [InlineData("", "hello", StringComparison.Ordinal, false)]
-        [InlineData("abcdefghijklmnopqrstuvwxyz", "abcdefghijklmnopqrstuvwxyz", StringComparison.Ordinal, true)]
-        [InlineData("abcdefghijklmnopqrstuvwxyz", "abcdefghijklmnopqrstuvwx", StringComparison.Ordinal, true)]
-        [InlineData("abcdefghijklmnopqrstuvwxyz", "abcdefghijklm", StringComparison.Ordinal, true)]
-        [InlineData("abcdefghijklmnopqrstuvwxyz", "ab_defghijklmnopqrstu", StringComparison.Ordinal, false)]
-        [InlineData("abcdefghijklmnopqrstuvwxyz", "abcdef_hijklmn", StringComparison.Ordinal, false)]
-        [InlineData("abcdefghijklmnopqrstuvwxyz", "abcdefghij_lmn", StringComparison.Ordinal, false)]
-        [InlineData("abcdefghijklmnopqrstuvwxyz", "a", StringComparison.Ordinal, true)]
-        [InlineData("abcdefghijklmnopqrstuvwxyz", "abcdefghijklmnopqrstuvwxyza", StringComparison.Ordinal, false)]
-        // OrdinalIgnoreCase
-        [InlineData("Hello", "Hel", StringComparison.OrdinalIgnoreCase, true)]
-        [InlineData("Hello", "Hello", StringComparison.OrdinalIgnoreCase, true)]
-        [InlineData("Hello", "Hello Larger", StringComparison.OrdinalIgnoreCase, false)]
-        [InlineData("Hello", "", StringComparison.OrdinalIgnoreCase, true)]
-        [InlineData("Hello", "HEL", StringComparison.OrdinalIgnoreCase, true)]
-        [InlineData("Hello", "Abc", StringComparison.OrdinalIgnoreCase, false)]
-        [InlineData("Hello", SoftHyphen + "Hel", StringComparison.OrdinalIgnoreCase, false)]
-        [InlineData("", "", StringComparison.OrdinalIgnoreCase, true)]
-        [InlineData("", "hello", StringComparison.OrdinalIgnoreCase, false)]
-        public static void StartsWith(string s, string value, StringComparison comparisonType, bool expected)
-        {
-            Assert.Equal(expected, s.AsSpan().StartsWith(value.AsSpan(), comparisonType));
-        }
-
-        [Theory]
-        [InlineData(StringComparison.Ordinal)]
-        [InlineData(StringComparison.OrdinalIgnoreCase)]
-        public static void StartsWith_NullInStrings(StringComparison comparison)
-        {
-            Assert.False("\0test".AsSpan().StartsWith("test".AsSpan(), comparison));
-            Assert.False("te\0st".AsSpan().StartsWith("test".AsSpan(), comparison));
-            Assert.True("te\0st".AsSpan().StartsWith("te\0s".AsSpan(), comparison));
-            Assert.True("test\0".AsSpan().StartsWith("test".AsSpan(), comparison));
-            Assert.False("test".AsSpan().StartsWith("te\0".AsSpan(), comparison));
-        }
-
         // NOTE: This is by design. Unix ignores the null characters (i.e. null characters have no weights for the string comparison).
         // For desired behavior, use ordinal comparison instead of linguistic comparison.
         // This is a known difference between Windows and Unix (https://github.com/dotnet/coreclr/issues/2051).
```

Comment:
What about other tests like StartsWith_NullInStrings_NonOrdinal? They were removed previously but not anymore: https://github.com/dotnet/corefx/pull/29378/files#diff-0ff06b1285d5667d42dc42f9df5ab694L346.

Reply:
@ahsonkhan some tests were "moved" to Span only file tests, they weren't present on string tests.On this new PR i removed only duplicates. To remove also these we need to add "string test part". If you want i could try to move STM "string like" test to common with string test addition, maybe on another PR.

Reply:
>  If you want i could try to move STM "string like" test to common with string test addition, maybe on another PR, it could take some time...

Separate PR would be great. Thanks for the contribution.